src/Qt-RSA/qrsaencryption.cpp

```diff
@@ -393,12 +398,47 @@ bool QRSAEncryption::checkSignMessage(const QByteArray &rawData, const QByteArra
 
     // hash, that was decrypt from recieved signature
     QByteArray recievedHash = decode(QByteArray::fromHex(rawData.mid(signStartPos + signMarkerLength, signLength)),
-                                     pubKey, BlockSize::OneByte);
+                                     pubKey, blockSizeMode);
 
     // if recievedHash == hashAlgorithm(recived message), then signed message is valid
     return recievedHash == QCryptographicHash::hash(message, HashAlgorithm::Sha256);
 }
 
+QByteArray QRSAEncryption::signMessageJava(QByteArray rawData, const QByteArray &privKey, const BlockSize blockSizeMode) {
+    QByteArray hash = QCryptographicHash::hash(rawData, HashAlgorithm::Sha256);
+
+    // ISO Object Identifier
+    QString digestOIDHexString ="3031300d060960864801650304020105000420";
+    QByteArray digestOID = QByteArray::fromHex(digestOIDHexString.toUtf8());
+    qDebug() << "digestOID: " << QString(digestOID.toHex());
+
+    // Padded
+    QString paddedHexString ="0001ffffffffffffffffffff00";
+    QByteArray padded = QByteArray::fromHex(paddedHexString.toUtf8());
+    qDebug() << "padded: " << QString(padded.toHex());
+
+    digestOID.append(hash);
+    padded.append(digestOID);
+    qDebug() << "encode: " << QString(padded.toHex());
+
+    QByteArray signature = encode(padded, privKey, blockSizeMode);
+    qDebug() << "signature: " << QString(signature.toHex());
+
+    //rawData.append(QString(SIGN_MARKER + signature.toHex() + SIGN_MARKER).toLatin1());
+
+    return signature;
+}
+bool QRSAEncryption::checkSignMessageJava(const QByteArray &message, const QByteArray &rawData, const QByteArray &pubKey, const BlockSize blockSizeMode) {
+    // start position of SIGN_MARKER in rawData
+    auto signStartPos = 19+ 13;
```

Comment:
Why sign raker began of this 19 + 13.
Please describe your code. 

If you want to found a beginning of your sign, use hooks (or magic symbol, you need to find it on array before processing).

Or use constant variables with readable names.